File: Backend/src/main/java/com/altairis_booking_system/Backend/service/TipoHabitacionService.java
package com.altairis_booking_system.Backend.service;

import com.altairis_booking_system.Backend.domain.TipoHabitacion;
import com.altairis_booking_system.Backend.dto.TipoHabitacionDTO;

import java.util.List;
import java.util.Optional;

public interface TipoHabitacionService {

    List<TipoHabitacionDTO> listadoTiposHabitaciones();

    Optional<TipoHabitacionDTO> buscarTipoHabitacionPorId(long id);

    TipoHabitacionDTO guardarTipoHabitacion(TipoHabitacionDTO tipoHabitacionDTO);

    Optional<TipoHabitacionDTO> actualiarTipoHabitacion(TipoHabitacionDTO tipoHabitacionDTO, long id);

    void eliminarTipoHabitacionPorId(long id);

    TipoHabitacionDTO convertirADTO(TipoHabitacion tipoHabitacion);

}

File: Frontend/src/app/services/tipo-habitacion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TipoHabitacion } from '../domain/TipoHabitacion';

@Injectable({
  providedIn: 'root'
})
export class TipoHabitacionService {

   private url: string = 'http://localhost:8080/api/tipos-habitaciones';

  constructor(private http: HttpClient) { 

  }

  listadoTiposHabitaciones(): Observable<TipoHabitacion[]>{
    return this.http.get<TipoHabitacion[]>(this.url);
  }
    
        
  buscarTipoHabitacionPorId(id: number): Observable<TipoHabitacion>{
    return this.http.get<TipoHabitacion>(`${this.url}/${id}`);
  }
    
  guardarTipoHabitacion(tipoHabitacion: TipoHabitacion){
    return this.http.post(this.url, tipoHabitacion);
  }

  editarTipoHabitacion(tipoHabitacion: TipoHabitacion) {
    return this.http.put(`${this.url}/${tipoHabitacion.idTipoHabitacion}`, tipoHabitacion);
  }

  eliminarTipoHabitacionPorId(idTipoHabitacion: number){
    return this.http.delete(`${this.url}/${idTipoHabitacion}`);
  }

}
